### PR TITLE
attempt to improve handler developer docs

### LINKED ISSUE #2954: Improve end-user-documentation for handler development
investigate what is needed

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

Added docs for handler development 

closes  #2954
closes https://github.com/inmanta/inmanta-core/issues/2534

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

**Comment by arnaudsjs** [comment]
@wouterdb The documentation does't compile. I get the error: `! LaTeX Error: Unknown graphics extension: .svg.`

File: docs/model_developers/handlers.rst [hunk 1]
```diff
@@ -72,24 +82,15 @@ The compiler collects all python modules from Inmanta modules that provide handl
...
-runs and reporting. The CRUDHandler handler also uses the result to determine whether create, delete
-or update needs to be invoked.
+Handlers should inherit the class :class:`~inmanta.agent.handler.CRUDHandler`. The
+:func:`@provider<inmanta.agent.handler.provider>` decorator registers the class with the orchestrator.
+
+Each Handler should override 4 methods of the CRUDHandler:
+
+1. :meth:`~inmanta.agent.handler.CRUDHandler.read_resource` to read the current state of the system.
+2. :meth:`~inmanta.agent.handler.CRUDHandler.create_resource` to create the resource if it doesn't exist.
+3. :meth:`~inmanta.agent.handler.CRUDHandler.update_resource` to update the resources when required.
```
Comment (arnaudsjs) [inline docs/model_developers/handlers.rst:92]:
```suggestion
3. :meth:`~inmanta.agent.handler.CRUDHandler.update_resource` to update the resource when required.
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]
See comments.

**Review by arnaudsjs — APPROVED** [review]

File: docs/Makefile [hunk 2]
```diff
@@ -10,9 +10,21 @@ BUILDDIR      = build
...
+# https://wiki.inkscape.org/wiki/Using_the_Command_Line#Deprecations_and_Replacements
+ifeq ($(shell inkscape --version | cut -d " " -f 2 | cut -d "." -f 1),0)
+SVG2PDF_FILE = -f
+SVG2PDF_PDF = -A
+SVG2PDF_PNG = -e
+else
+SVG2PDF_FILE =
+SVG2PDF_PDF = --export-type=pdf -o
+SVG2PDF_PNG = --export-type=png -o
+endif
```
Comment (sanderr) [inline docs/Makefile:26]: This will make my life easier.

**Review by sanderr — APPROVED** [review]

File: docs/Makefile (hunk 2 above)
Comment (arnaudsjs) [reply docs/Makefile:26]: If we apply this, we should close https://github.com/inmanta/inmanta-core/issues/2534 as well.

**Pull request closed without merging**